Sources/SymbolKit/SymbolGraph/SymbolGraph+Overloads.swift

```diff
@@ -117,13 +152,47 @@ extension SymbolGraph.Symbol {
     public var isOverloadGroup: Bool {
         self.identifier.precise.hasSuffix(Self.overloadGroupIdentifierSuffix)
     }
+
+    /// Returns a sort comparator suitable for sorting symbols within an overload group.
+    ///
+    /// The sort order for overloads follows the following algorithm:
+    /// - If `orderByDeclaration` is false, symbols are sorted by their precise identifier.
+    /// - If `orderByDeclaration` is true, symbols are sorted by their ``DeclarationFragments``
+    ///   mixin, falling back to their precise identifier if their declarations are equal.
+    internal static func sortForOverloads(orderByDeclaration: Bool) -> ((Self, Self) -> Bool) {
+        return { isDeclarationBefore(
+            $0, $1, orderByDeclaration: orderByDeclaration,
+            getDeclaration: { $0.declarationFragments!.rendered },
+            getIdentifier: { $0.identifier.precise }
+        )}
+    }
 }
 
-fileprivate extension [SymbolGraph.Symbol.DeclarationFragments.Fragment] {
-    static func < (lhs: Self, rhs: Self) -> Bool {
-        let renderedLhs = lhs.map(\.spelling).joined()
-        let renderedRhs = rhs.map(\.spelling).joined()
+extension UnifiedSymbolGraph.Symbol {
+    /// If a symbol has declaration fragments mixins, returns the one that lexicographically sorts first.
+    var firstAvailableDeclaration: String? {
+        let uniqueDeclarations = Set(declarationFragments.values.map(\.rendered))
+        return uniqueDeclarations.sorted().first
+    }
+
+    /// Returns a sort comparator suitable for sorting symbols within an overload group.
+    ///
+    /// The sort order for overloads follows the following algorithm:
+    /// - If `orderByDeclaration` is false, symbols are sorted by their unique identifier.
+    /// - If `orderByDeclaration` is true, symbols are sorted by their declaration (as returned by
+    ///   ``firstAvailableDeclaration``), falling back to their unique identifier if their
+    ///   declarations are equal.
+    internal static func sortForOverloads(orderByDeclaration: Bool) -> ((UnifiedSymbolGraph.Symbol, UnifiedSymbolGraph.Symbol) -> Bool) {
+        return { isDeclarationBefore(
+            $0, $1, orderByDeclaration: orderByDeclaration,
+            getDeclaration: { $0.firstAvailableDeclaration! },
+            getIdentifier: { $0.uniqueIdentifier }
+        )}
+    }
+}
 
-        return renderedLhs < renderedRhs
+internal extension [SymbolGraph.Symbol.DeclarationFragments.Fragment] {
```

Comment:
> Isn't `internal` the default? Can we drop that keyword here?

It is the default, but i like being explicit when the keyword is available. 😅 I think @d-ronnqvist also called out the use of `internal` in a previous PR, so i can drop them here.

> Also, I haven't seen this style before: Is this equivalent to [...]

Yes! I think this syntax was added in one of the Swift 5 point releases, from around the time i was starting to use it.

> Which is preferred from a Swift style point of view?

That one i don't know. I very much like this sugar since you almost never write the `Array` type name directly and this makes it much easier to understand. But i'll defer to @d-ronnqvist for the call of what is more "Swifty".